src/warnings.js

```diff
@@ -54,12 +54,12 @@ export const warnings = Object.freeze({
 });
 
 export function generateWarning(kind, options) {
-  const { location, file = null, value = null } = options;
+  const { location, file = null, value = null, source = "JS-X-Ray"} = options;
```

Comment:
Should we validate source ?
```js
const kWarningsSource = new Set(["JS-X-Ray", "Scanner"]);
// [...]
if (!kWarningsSource.has(source) {
  throw new Error(`Source ${source} is not allowed`);
}
```

Reply:
Sorry, I hadn't seen that before my last commit. I've finally removed the union type (should I leave it in place?).

Reply:
I think it's ok to not check the source then 👍 
However, i would not set it as `null` by default but `JS-X-Ray` instead.

Reply:
OK noted

Reply:
I dont think we need to validate source (could limit customization of source in future).